tasks/gulp/compile-assets.js

```diff
@@ -97,18 +94,18 @@ gulp.task('scss:compile', () => {
 
 // Compile js task for preview ----------
 // --------------------------------------
-gulp.task('js:compile', () => {
-  return gulp.src([
-    '!' + configPaths.src + '**/*.test.js',
-    configPaths.src + '**/*.js'
-  ])
-    .pipe(concat('govuk-frontend.js'))
-    .pipe(gulpif(isDist, uglify()))
-    .pipe(gulpif(isDist,
-      rename({
-        extname: '.min.js'
-      })
-    ))
-    .pipe(eol())
-    .pipe(gulp.dest(taskArguments.destination + '/js/'))
+gulp.task('js:compile', (callback) => {
+  const { exec } = require('child_process')
+  const output = taskArguments.destination
+  const outputEnvironment = output ? `OUTPUT=${output} ` : ''
+  exec(`${outputEnvironment} node_modules/.bin/webpack --config config/webpack.config.js`, (err, stdout, stderr) => {
```

Comment:
@alex-ju Would you be able to add a comment here along the lines of what you explained to me about why we need to execute Webpack inside gulp?